### ISSUE TITLE
Ignore emacs lock files

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#637]
At least on some platforms, Emacs creates lock files that are symlinks using the name of the file being edited prefixed by ".#". These files only exist while the buffer containing the file has unsaved modifications.

Since these files match the extension for documentation sources (whether .rst or .txt), Sphinx complains that there's a documentation file that's not mentioned in the toctree using a traceback:

```
Exception occurred:
  File "/home/fdrake/env/dot_buildout/eggs/docutils-0.7-py2.6.egg/docutils/io.py
", line 234, in __init__
    self.source = open(source_path, mode, **kwargs)
IOError: [Errno 2] No such file or directory: '/home/fdrake/.../.#sites.rst'
```

Sphinx should probably be trained to ignore these symlinks.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/637
- Originally reported by: Fred Drake
- Originally created at: 2011-02-15T17:16:27.044

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 17:22:06+00:00_

Fixed in <<cset d05588da413ba9bb21445f0a33740a3502bad83a>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 17:27:25+00:00_

Fix #637: ignore strange Emacs lock files by default.

→ <<cset d05588da413ba9bb21445f0a33740a3502bad83a>>

**Comment by shimizukawa** [comment]
_From Anonymous on 2012-05-07 11:29:34+00:00_

I still get this error in Sphinx 1.1.3:

{{{
    Running Sphinx v1.1.3
    loading pickled environment... failed: env version not current
    building [html]: targets for 14 source files that are out of date
    updating environment: 14 added, 0 changed, 0 removed
    reading sources... [ 28%] gc3apps/.#grosetta  
    Exception occurred:
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/docutils/io.py", line 217, in **init**
        self.source = open(source_path, mode, **kwargs)
    IOError: [Errno 2] No such file or directory: '/home/rmurri/gc3/gc3pie.googlecode.com/gc3pie/docs/gc3apps/.#grosetta.txt'
}}}

Here's the full traceback for reference:
{{{
    python /home/rmurri/.virtualenvs/gc3pie-dev/bin/sphinx-build -b html -d ./doctrees -D latex_paper_size=a4  . ./html
    # Sphinx version: 1.1.3
    # Python version: 2.7.3
    # Docutils version: 0.8.1 release
    # Jinja2 version: 2.6
    Traceback (most recent call last):
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/cmdline.py", line 189, in main
        app.build(force_all, filenames)
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/application.py", line 204, in build
        self.builder.build_update()
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/builders/**init**.py", line 196, in build_update
        'out of date' % len(to_build))
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/builders/**init**.py", line 216, in build
        purple, length):
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/builders/**init**.py", line 120, in status_iterator
        for item in iterable:
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/environment.py", line 613, in update_generator
        self.read_doc(docname, app=app)
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/environment.py", line 758, in read_doc
        pub.set_source(None, src_path.encode(fs_encoding))
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/docutils/core.py", line 177, in set_source
        encoding=self.settings.input_encoding)
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/sphinx/environment.py", line 732, in **init**
        FileInput.**init**(self_, _args, *_kwds)
      File "/home/rmurri/.virtualenvs/gc3pie-dev/local/lib/python2.7/site-packages/docutils/io.py", line 217, in **init**
        self.source = open(source_path, mode, **kwargs)
    IOError: [Errno 2] No such file or directory: '/home/rmurri/gc3/gc3pie.googlecode.com/gc3pie/docs/gc3apps/.#grosetta.txt'
}}}
